apiserver/facades/client/sshclient/facade_test.go

```diff
@@ -441,6 +443,72 @@ func (s *facadeSuite) assertModelCredentialForSSH(c *gc.C, f func(authorizer *mo
 	})
 }
 
+func (s *facadeSuite) TestGetVirtualHostnameForEntity(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+	backend := mocks.NewMockBackend(ctrl)
+	authorizer := mocks.NewMockAuthorizer(ctrl)
+	broker := mocks.NewMockBroker(ctrl)
+
+	backend.EXPECT().ModelTag().Return(testing.ModelTag).AnyTimes()
+	backend.EXPECT().ControllerTag().Return(testing.ControllerTag).AnyTimes()
+
+	authorizer.EXPECT().AuthClient().Return(true).AnyTimes()
+	authorizer.EXPECT().HasPermission(permission.SuperuserAccess, testing.ControllerTag).Return(authentication.ErrorEntityMissingPermission).AnyTimes()
+	authorizer.EXPECT().HasPermission(permission.AdminAccess, testing.ModelTag).Return(nil).AnyTimes()
```

Comment:
Shouldn't we rather do exact number of time, rather than `AnyTimes()`?

Reply:
Ah but we're using them in the table test. Is it worthwhile to create a separate test to verify the auth?